warehouse/ingest-core/src/main/java/datawave/iterators/FrequencyColumnIterator.java

```diff
@@ -0,0 +1,179 @@
+package datawave.iterators;
+
+import datawave.data.ColumnFamilyConstants;
+import datawave.query.util.Frequency;
+import datawave.query.util.FrequencyFamilyCounter;
+import datawave.query.util.MetadataHelper;
+import datawave.query.util.YearMonthDay;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.user.TransformingIterator;
+import org.apache.hadoop.io.Text;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;;
+import java.util.Map;
+
+public class FrequencyColumnIterator extends TransformingIterator {
+    
+    private FrequencyFamilyCounter frequencyFamilyCounter;
```

Comment:
We should not have a global state here.  All transformation should be in the context of the transformRange method and returned via the kvBuffer parameter.  We can remove the frequencyFamilyCounter and the rowIdToCompressFreqCQMap class members.